src/app/3d/qgs3dmapcanvaswidget.cpp

```diff
@@ -62,6 +63,9 @@
 
 #include "moc_qgs3dmapcanvaswidget.cpp"
 
+const QgsSettingsEntryDouble *Qgs3DMapCanvasWidget::settingClippingTolerance = new QgsSettingsEntryDouble( QStringLiteral( "tolerance" ), QgsSettingsTree::sTree3DMap, 100, QStringLiteral( "Tolerance distance for 3D Map cross section" ), Qgis::SettingsOptions(), 0 );
+const QgsSettingsEntryBool *Qgs3DMapCanvasWidget::settingDynamicClipping = new QgsSettingsEntryBool( QStringLiteral( "enable-dynamic-clipping" ), QgsSettingsTree::sTree3DMap, true, QStringLiteral( "Whether dynamic clipping is set for 3D cross section" ) );
```

Comment:
I would suggest to rename "dynamic clipping" as it is unclear what that means... for the label, how about "Cross section tolerance lock - whether the tolerance is set by mouse every time a new cross section is picked"